lib/pp.mli

```diff
@@ -75,6 +75,9 @@ val ismt  : t -> bool
 
 val comment  : string list -> t
 
+val fmt : ('a, unit, t) format -> 'a
+(** complex formatting *)
```

Comment:
This seems very useful, I've often missed something like that. In fact, I often wonder why Rocq uses the `Pp` module instead of just relying on `Format`-based stuff.  Is there any good reason not to transition to something like that in the future?

Reply:
Not sure how important this is in practice, but Pp is a static datatype (no closures) which can be progressively built.
For instance it allows to print an error message (using the global state), reset the state, then finish printing by doing essentially `printf "Error:@ @[%s@]" msg` (but if you do that actual printf, msg is just a string so the box does nothing)
It's also serializable to send to ides, for instance coq-lsp and vsrocq both do some custom printing instead of using the Format engine AFAIK

Reply:
Also CoqIDE sends Pp.t on the wire. But indeed, Rocq has a need for a pretty printing document type, so far we are just using `Pp.t` which is a straightfoward reifycation of Format's API, but that falls short for many cases.

We have been experimenting with some different formats in https://github.com/ejgallego/coq-lsp/pull/668

Reply:
Reacting to the static dataype, since `5.3`, the compiler also has its own reified representation of `Format` streams https://ocaml.org/manual/5.3/api/compilerlibref/Format_doc.html which could be made more broadly available as a separate library.